components/director/internal/domain/apptemplate/service_test.go

```diff
@@ -0,0 +1,408 @@
+package apptemplate_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/apptemplate"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/apptemplate/automock"
+	"github.com/kyma-incubator/compass/components/director/internal/tenant"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestService_Create(t *testing.T) {
+	// GIVEN
+	ctx := tenant.SaveToContext(context.TODO(), testTenant)
+	uidSvcFn := func() *automock.UIDService {
+		uidSvc := &automock.UIDService{}
+		uidSvc.On("Generate").Return(testID).Once()
+		return uidSvc
+	}
+	modelAppTemplate := fixModelAppTemplate(testID, testName)
+
+	testCases := []struct {
+		Name              string
+		Input             *model.ApplicationTemplateInput
+		AppTemplateRepoFn func() *automock.ApplicationTemplateRepository
+		ExpectedError     error
+		ExpectedOutput    string
+	}{
+		{
+			Name:  "Success",
+			Input: fixModelAppTemplateInput(testName, fixApplicationCreateInputString()),
+			AppTemplateRepoFn: func() *automock.ApplicationTemplateRepository {
+				appTemplateRepo := &automock.ApplicationTemplateRepository{}
+				appTemplateRepo.On("Create", ctx, *modelAppTemplate).Return(nil).Once()
+				return appTemplateRepo
+			},
+			ExpectedOutput: testID,
+		},
+		{
+			Name: "Error when application template placeholders are not unique",
+			Input: &model.ApplicationTemplateInput{
+				Placeholders: []model.ApplicationTemplatePlaceholder{
+					{
+						Name:        testName,
+						Description: nil,
+					},
+					{
+						Name:        testName,
+						Description: nil,
+					},
+				},
+			},
+			AppTemplateRepoFn: func() *automock.ApplicationTemplateRepository {
+				appTemplateRepo := &automock.ApplicationTemplateRepository{}
+				return appTemplateRepo
+			},
+			ExpectedError: errors.New("while validating uniqueness of placeholders"),
+		},
+		{
+			Name:  "Error when creating",
```

Comment:
name of this test will be:
`TestService_Create/Error_when_creating`
not good, not terrible